src/generate/peripheral.rs

```diff
@@ -679,7 +679,10 @@ fn expand_register(
     match register {
         Register::Single(info) => register_expanded.push(RegisterBlockField {
             field: convert_svd_register(register, name),
-            description: info.description.clone().unwrap(),
+            description: info
+                .description
+                .clone()
+                .unwrap_or("No description available".to_string()),
```

Comment:
Maybe it's better to use `unwrap_or_else` in such cases to not create a `String` object every time.

Reply:
There are two functions with pretty the same purpose: `unwrap_or` and `unwrap_or_else`. `unwrap_or` takes a value (and you need to construct it) every time, `unwrap_or_else` constructs this value only if needed by calling a function.

Reply:
The `"No description available".to_string()` will always be executed independently of whether `info.description` is `None` or `Some(...)`. If the description is `None` then the result of the operation will be dropped so you're constructing the same string over and over again just to drop it a moment later.

Reply:
Oh I see. I didnt know that. Thanks for the pointer. I'll make the change

Reply:
There's a clippy lint about that: https://rust-lang.github.io/rust-clippy/master/index.html#or_fun_call